Modules/Filtering/ImageGrid/test/itkBSplineScatteredDataPointSetToImageFilterTest5.cxx

```diff
@@ -115,19 +112,46 @@ itkBSplineScatteredDataPointSetToImageFilterTest5(int argc, char * argv[])
   close.Fill(1);
   filter->SetCloseDimension(close);
 
-
   ITK_TRY_EXPECT_NO_EXCEPTION(filter->Update());
 
-  // Cast the PhiLattice
-  using CastImageFilterType = itk::CastImageFilter<FilterType::PointDataImageType, OutputImageType>;
-  auto caster = CastImageFilterType::New();
-  caster->SetInput(filter->GetPhiLattice());
+  // using OutputPixelType = unsigned char;
+  using OutputPixelType = uint8_t;
+  using OutputImageType = itk::Image<OutputPixelType, ParametricDimension>;
+
+  auto                     phiLatticeImage = filter->GetPhiLattice();
+  OutputImageType::Pointer outputImage = OutputImageType::New();
+  outputImage->CopyInformation(phiLatticeImage);
+  outputImage->SetRegions(phiLatticeImage->GetBufferedRegion());
+  outputImage->Allocate(true);
+
+  // phiLatticeImage contains pixels of type Vector<double, 3>
+  // with minmax values of (-5, 5) and magnitude max of about 5
+  auto myiterator =
+    itk::ImageRegionIterator<FilterType::PointDataImageType>(phiLatticeImage, phiLatticeImage->GetBufferedRegion());
+  auto myOutiterator = itk::ImageRegionIterator<OutputImageType>(outputImage, outputImage->GetBufferedRegion());
 
+  float max_magnitude_value = 0.0;
+  while (!myiterator.IsAtEnd())
+  {
+    // Convert vector image to magnitude and scale to use range of png values
+    const auto magvalue = myiterator.Value().GetNorm();
+    myOutiterator.Set(magvalue);
```

Comment:
Doesn't this create weird aliasing artifacts? E.g. if magnitudes are in [-4.5,4.4] range, they will get aliased to just five values (0,1,2,3,4). Later scaling to 0-255 will transform them into (0,56,113,170,226). If we want to avoid recomputing the norms twice, we need an intermediate float image. Also, using a single iterator is single-threaded. Maybe not important for a test?

Reply:
Good catch @dzenanz .  This computation in the test is very fast anyway.  I don't think it is worth doing performance testing.  The entire test only takes 150ms as is.

New patch coming shortly.